agenta-cli/agenta/sdk/agenta.py

```diff
@@ -151,46 +157,85 @@ def wrapper(*args, **kwargs):
             return result
         except Exception as e:
             if sys.version_info.major == 3 and sys.version_info.minor < 10:
-                traceback_str = ''.join(traceback.format_exception(None, e, e.__traceback__))
+                traceback_str = "".join(
+                    traceback.format_exception(None, e, e.__traceback__)
+                )
             else:
-                traceback_str = ''.join(traceback.format_exception(e, value=e, tb=e.__traceback__))
-            return JSONResponse(status_code=500, content={"error": str(e), "traceback": traceback_str})
+                traceback_str = "".join(
+                    traceback.format_exception(e, value=e, tb=e.__traceback__)
+                )
+            return JSONResponse(
+                status_code=500,
+                content={"error": str(e), "traceback": traceback_str},
+            )
+
     new_params = []
+    instances_to_override = []
     for name, param in sig.parameters.items():
         if name in app_params:
-            new_params.append(
-                inspect.Parameter(
-                    name,
-                    inspect.Parameter.KEYWORD_ONLY,
-                    default=Body(app_params[name]),
-                    annotation=Optional[param.annotation]
+            if param.annotation is MultipleChoiceParam:
```

Comment:
can you add some comment explaining what you are doing in the case of MultipleChoiceParam and why for the future